### PR TITLE
[REVIEW] Add fluent builder apis to `json_reader_options` and `avro_reader_options`

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
#5962

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Review by mike-wendt — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]
This looks good.

**Comment by rgsl888prabhu** [comment]
rerun tests

File: python/cudf/cudf/_lib/avro.pyx [hunk 1]
```diff
@@ -21,35 +25,32 @@ cpdef read_avro(datasource, columns=None, skip_rows=-1, num_rows=-1):
...
+        c_columns.reserve(len(columns))
+        for col in columns:
+            c_columns.push_back(str(col).encode())
+
+    cdef avro_reader_options options = move(
+        avro_reader_options.builder(make_source_info([datasource])).
+        columns(c_columns).
+        skip_rows(<size_type> skip_rows).
+        num_rows(<size_type> num_rows).
+        build()
```
Comment (kkraus14) [inline python/cudf/cudf/_lib/avro.pyx:46]:
```suggestion
        avro_reader_options.builder(make_source_info([datasource]))
        .columns(c_columns)
        .skip_rows(<size_type> skip_rows)
        .num_rows(<size_type> num_rows)
        .build()
```

File: python/cudf/cudf/_lib/json.pyx [hunk 2]
```diff
@@ -40,42 +46,56 @@ cpdef read_json(filepath_or_buffer, dtype,
...
-    args.byte_range_offset = c_range_offset
-    args.byte_range_size = c_range_size
+    cdef json_reader_options opts = move(
+        json_reader_options.builder(make_source_info([path_or_data])).
+        dtypes(c_dtypes).
+        compression(c_compression).
+        lines(c_lines).
+        byte_range_offset(c_range_offset).
+        byte_range_size(c_range_size).
+        build()
```
Comment (kkraus14) [inline python/cudf/cudf/_lib/json.pyx:91]:
```suggestion
        json_reader_options.builder(make_source_info([path_or_data]))
        .dtypes(c_dtypes)
        .compression(c_compression)
        .lines(c_lines)
        .byte_range_offset(c_range_offset)
        .byte_range_size(c_range_size)
        .build()
```

**Review by kkraus14 — CHANGES_REQUESTED** [review]

**Review by nvdbaranec — CHANGES_REQUESTED** [review]

**Review by kkraus14 — APPROVED** [review]

File: cpp/include/cudf/io/readers.hpp [hunk 3]
```diff
@@ -44,175 +44,6 @@ namespace cudf {
 //! IO interfaces
 namespace io {
 namespace detail {
```
[the remark this answers is not in the document]
Comment (rgsl888prabhu) [reply cpp/include/cudf/io/readers.hpp:46]: These files will be removed at the end of refactoring, so doesn't matter.

**Review by nvdbaranec — APPROVED** [review]

**Pull request merged**